diff -r 2cc595391006 experimental/dddvb/ddbridge/ddbridge-core.c
--- linux/drivers/media/pci/ddbridge/ddbridge-core.c.old	Tue Nov 05 08:03:36 2013 +0100
+++ linux/drivers/media/pci/ddbridge/ddbridge-core.c	Tue Nov 05 09:18:10 2013 +0100
@@ -943,16 +943,19 @@
 	return status;
 }
 
-#ifdef CONFIG_DVB_DRXK
+#if IS_ENABLED(CONFIG_DVB_DRXK)
 static int demod_attach_drxk(struct ddb_input *input)
 {
 	struct i2c_adapter *i2c = &input->port->i2c->adap;
 	struct ddb_dvb *dvb = &input->port->dvb[input->nr & 1];
 	struct dvb_frontend *fe;
-
-	fe = dvb->fe = dvb_attach(drxk_attach,
-				  i2c, 0x29 + (input->nr & 1),
-				  &dvb->fe2);
+	struct drxk_config config;
+
+	memset(&config, 0, sizeof(config));
+	config.adr = 0x29 + (input->nr & 1);
+	config.microcode_name = "drxk_a3.mc";
+
+	fe = dvb->fe = dvb_attach(drxk_attach, &config, i2c);
 	if (!fe) {
 		pr_err("No DRXK found!\n");
 		return -ENODEV;
@@ -1872,7 +1875,7 @@
 		if (tuner_attach_stv6111(input, 1) < 0)
 			return -ENODEV;
 		break;
-#ifdef CONFIG_DVB_DRXK
+#if IS_ENABLED(CONFIG_DVB_DRXK)
 	case DDB_TUNER_DVBCT_TR:
 		if (demod_attach_drxk(input) < 0)
 			return -ENODEV;
