--- linux/drivers/media/pci/ddbridge/ddbridge.c.org	2015-01-21 00:28:11.304747939 +0100
+++ linux/drivers/media/pci/ddbridge/ddbridge.c	2015-01-21 00:31:24.152742978 +0100
@@ -158,23 +158,30 @@ static int __devinit ddb_probe(struct pc
 
 #ifdef CONFIG_PCI_MSI
 	if (msi && pci_msi_enabled()) {
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 16, 0))
-		stat = pci_enable_msi_range(dev->pdev, 2, 2);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 15, 0))
+		stat = pci_enable_msi_range(dev->pdev, 1, 2);
+		if (stat >= 1) {
+			dev->msi = stat;
+			pr_info("DDBridge using %d MSI interrupt(s)\n", dev->msi);
+			irq_flag = 0;
+		} else
+			pr_info("DDBridge: MSI not available.\n");
+
 #else
 		stat = pci_enable_msi_block(dev->pdev, 2);
-#endif
 		if (stat == 0) {
 			dev->msi = 1;
-			pr_info("DDBrige using 2 MSI interrupts\n");
+			pr_info("DDBridge using 2 MSI interrupts\n");
 		}
 		if (stat == 1)
 			stat = pci_enable_msi(dev->pdev);
 		if (stat < 0) {
-			pr_info(": MSI not available.\n");
+			pr_info("DDBridge: MSI not available.\n");
 		} else {
 			irq_flag = 0;
 			dev->msi++;
 		}
+#endif
 	}
 	if (dev->msi == 2) {
 		stat = request_irq(dev->pdev->irq, irq_handler0,
