--- linux/drivers/media/pci/ddbridge/ddbridge-core.c.orig	2015-01-23 20:05:44.707853587 +0100
+++ linux/drivers/media/pci/ddbridge/ddbridge-core.c	2015-01-23 20:30:29.755809164 +0100
@@ -1849,7 +1849,8 @@ static int dvb_input_attach(struct ddb_i
 	dvb->fe = dvb->fe2 = 0;
 	switch (port->type) {
 	case DDB_TUNER_MXL5XX:
-		fe_attach_mxl5xx(input);
+		if (fe_attach_mxl5xx(input) < 0)
+			return -ENODEV;
 		break;
 	case DDB_TUNER_DVBS_ST:
 		if (demod_attach_stv0900(input, 0) < 0)
