--- linux/drivers/media/pci/ddbridge/ddbridge-core.c.orig	2014-11-01 16:54:38.811103608 +0100
+++ linux/drivers/media/pci/ddbridge/ddbridge-core.c	2014-11-01 18:43:16.783247581 +0100
@@ -3585,16 +3585,16 @@ static struct device_attribute ddb_attrs
 	__ATTR_RO(ports),
 	__ATTR_RO(ts_irq),
 	__ATTR_RO(i2c_irq),
-	__ATTR(gap0, 0666, gap_show, gap_store),
-	__ATTR(gap1, 0666, gap_show, gap_store),
-	__ATTR(gap2, 0666, gap_show, gap_store),
-	__ATTR(gap3, 0666, gap_show, gap_store),
+	__ATTR(gap0, 0664, gap_show, gap_store),
+	__ATTR(gap1, 0664, gap_show, gap_store),
+	__ATTR(gap2, 0664, gap_show, gap_store),
+	__ATTR(gap3, 0664, gap_show, gap_store),
 	__ATTR_RO(hwid),
 	__ATTR_RO(regmap),
 #if 0
 	__ATTR_RO(qam),
 #endif
-	__ATTR(redirect, 0666, redirect_show, redirect_store),
+	__ATTR(redirect, 0664, redirect_show, redirect_store),
 	__ATTR_MRO(snr,  bsnr_show),
 	__ATTR_NULL,
 };
@@ -3617,21 +3617,21 @@ static struct device_attribute ddb_attrs
 };
 
 static struct device_attribute ddb_attrs_fan[] = {
-	__ATTR(fan, 0666, fan_show, fan_store),
+	__ATTR(fan, 0664, fan_show, fan_store),
 };
 
 static struct device_attribute ddb_attrs_snr[] = {
-	__ATTR(snr0, 0666, snr_show, snr_store),
-	__ATTR(snr1, 0666, snr_show, snr_store),
-	__ATTR(snr2, 0666, snr_show, snr_store),
-	__ATTR(snr3, 0666, snr_show, snr_store),
+	__ATTR(snr0, 0664, snr_show, snr_store),
+	__ATTR(snr1, 0664, snr_show, snr_store),
+	__ATTR(snr2, 0664, snr_show, snr_store),
+	__ATTR(snr3, 0664, snr_show, snr_store),
 };
 
 static struct device_attribute ddb_attrs_led[] = {
-	__ATTR(led0, 0666, led_show, led_store),
-	__ATTR(led1, 0666, led_show, led_store),
-	__ATTR(led2, 0666, led_show, led_store),
-	__ATTR(led3, 0666, led_show, led_store),
+	__ATTR(led0, 0664, led_show, led_store),
+	__ATTR(led1, 0664, led_show, led_store),
+	__ATTR(led2, 0664, led_show, led_store),
+	__ATTR(led3, 0664, led_show, led_store),
 };
 
 static struct class ddb_class = {
