--- linux/drivers/media/dvb-frontends/mxl5xx.c.orig	2015-01-24 21:01:07.318729957 +0100
+++ linux/drivers/media/dvb-frontends/mxl5xx.c	2015-01-24 21:02:01.662731271 +0100
@@ -468,7 +468,7 @@ static int tune(struct dvb_frontend *fe,
 	r = read_status(fe, status);
 	if (r)
 		return r;
-
+/*
 	if (*status & FE_HAS_LOCK)
 		return 0;
 
@@ -477,7 +477,7 @@ static int tune(struct dvb_frontend *fe,
 	else
 		p->delivery_system = SYS_DVBS;
 	set_parameters(fe);
-
+*/
 	return 0;
 }
 
