--- todo/drivers/media/dvb-core/dvb_frontend.h	2014-08-23 05:45:35.000000000 +0200
+++ experimental/dddvb/dvb-core/dvb_frontend.h	2015-01-24 18:48:34.094537651 +0100
@@ -306,6 +304,7 @@ struct dvb_frontend_ops {
 	int (*i2c_gate_ctrl)(struct dvb_frontend* fe, int enable);
 	int (*ts_bus_ctrl)(struct dvb_frontend* fe, int acquire);
 	int (*set_lna)(struct dvb_frontend *);
+	int (*set_input)(struct dvb_frontend *, int);
 
 	/* These callbacks are for devices that implement their own
 	 * tuning algorithms, rather than a simple swzigzag
@@ -317,6 +316,8 @@ struct dvb_frontend_ops {
 
 	int (*set_property)(struct dvb_frontend* fe, struct dtv_property* tvp);
 	int (*get_property)(struct dvb_frontend* fe, struct dtv_property* tvp);
+
+	u8 xbar[3];
 };
 
 #ifdef __DVB_CORE__
@@ -395,7 +396,8 @@ struct dtv_frontend_properties {
 	u8			atscmh_sccc_code_mode_d;
 
 	u32			lna;
-
+	s32                     input;
+	
 	/* statistics data */
 	struct dtv_fe_stats	strength;
 	struct dtv_fe_stats	cnr;
