--- linux/drivers/media/pci/ttpci/av7110_ipack.c.old	2012-08-14 05:45:22.000000000 +0200
+++ linux/drivers/media/pci/ttpci/av7110_ipack.c	2013-01-27 14:41:24.114810675 +0100
@@ -241,9 +241,18 @@ int av7110_ipack_instant_repack (const u
 
 		case 8:
 			if (!p->done && p->mpeg == 2) {
+				int i;
 				p->hlength = buf[c];
 				c++;
 				p->found++;
+				/* discard non-MPEG-2 video, i.e. H264 */
+				i = p->found + p->hlength;
+				if (count > i + 2 && VIDEO_STREAM_S <= p->cid && p->cid <= VIDEO_STREAM_E &&
+				    !(buf[i] == 0x00 && buf[i+1] == 0x00 && buf[i+2] == 0x01)) {
+					/* not MPEG-2 */
+					av7110_ipack_reset(p);
+				}
+
 			}
 			break;
 		}
