--- linux/drivers/media/Kconfig.old	2014-03-05 20:18:24.487948560 +0100
+++ linux/drivers/media/Kconfig	2014-03-05 20:22:21.503951925 +0100
@@ -2,6 +2,12 @@
 # Multimedia device configuration
 #
 
+# required because of brain-damaged mconf behaviour
+config BULLSHIT
+        bool
+        option modules
+        default y
+
 menuconfig MEDIA_SUPPORT
 	tristate "Multimedia support"
 	depends on HAS_IOMEM
